File: examples/init_lib.py
#<![CDATA[
import sys
from RASCH import *

# initialize libRASCH
ra = RASCH()
if not ra:
    print "can't initialize libRASCH"
    sys.exit()
[err_num, err_text] = ra.get_error()
if err_num != 1:
    print "while initializing libRASCH, error #%d occured:\n " \
          "%s\n" % err_num, err_text
    sys.exit()
    
# get some infos
value = ra.get_info(info='num_plugins')
if (value.is_ok()):
    print value.name(), "("+value.desc()+"):", value.value()

# find all measurements in a directory
meas = ra.find_meas(sys.argv[1])
print "measurements found in " + sys.argv[1] + ":\n"
cnt = 1
for item in meas:
    print "%d: %s" % (cnt, item.filename())
    cnt = cnt + 1

#]]>

File: examples/init_lib.pl
#<![CDATA[
use strict;
use RASCH;

# initialize libRASCH
my $ra = new RASCH or die "error initializing libRASCH\n";
my ($err_num, $err_text) = $ra->get_error ();
if ($err_num != 1)
{
    print "while initializing libRASCH, error # $err_num " .
        "occured:\n $err_text\n";
    exit -1;
}

# get some infos
my $value = $ra->get_info(info =>'num_plugins');
if ($value->is_ok())
{
    print $value->name() . " (" . $value->desc() . "): " . $value->value() . "\n";
}

# find all measurements in a directory
my $meas = $ra->find_meas($ARGV[0]);
print "measurements found in $ARGV[0]:\n";
my $cnt = 1;
for (@$meas)
{
    print "$cnt: " . $_->filename() . "\n";
    $cnt++;
}

# ra_close() will be called when $ra is being destroyed

exit 0;
#]]>
